sustaingym/envs/building/stochastic_uncontrollable_generator.py

```diff
@@ -0,0 +1,222 @@
+from __future__ import annotations
+
+from collections.abc import Iterable
+
+import numpy as np
+import pandas as pd
+import scipy
+
+
+class StochasticUncontrollableGenerator():
+    def __init__(self, num_episodes: int | None = None):
+        """
+        Initializes a generator class for the uncontrollable features in BuildingEnv.
+
+        Args:
+            num_episodes: The number of episodes in the provided observation data. If
+                given, instance will treat each row in the observation data as a separate
+                raw dataset to split into seasons and fit distributions to. If None,
+                will use the provided raw data as a single episode.
+        """
+        self.episodes = num_episodes
+
+        self.observations = [[]]
+
+        self.summer_observations = []
+        self.winter_observations = []
+
+        self.summer_dists = None
+        self.winter_dists = None
+        
+        self.block_size = 0
+
+    def split_observations_into_seasons(
+        self, observation_data: np.ndarray
+    ) -> (np.ndarray, np.ndarray):
+        """
+        Splits observation data into summer and winter seasons.
+
+        Args:
+            observation_data (num_episodes x len_of_episode, num_features): 
+                The collected observation data.
+        
+        Returns:
+            summer_observations (len_of_season x num_episodes, num_features): 
+                Summer ambient features.
+            winter_observations (len_of_season x num_episodes, num_features): 
+                Winter ambient features.
```

Comment:
For documenting function parameters and return values, you MUST follow the following format in order for the automatic documentation generation (by Sphinx) to work:

```python
def f(x: np.ndarray) -> tuple[np.ndarray, np.ndarray]:
    """
    Description of function.

    Args:
        x: description

    Returns:
        return_value1: description
        return_value2: description
    """
```

You can't have `x (shape): description`. The colon must immediately follow the argument name, so you should put the shape information in the variable description